core/templates/dev/head/pages/exploration_player/exploration_player.html

```diff
@@ -6,21 +6,18 @@
 
 {% block header_js %}
   {{ super() }}
-  {% if exploration_version %}
-    <script type="text/javascript">
-      GLOBALS.collectionId = JSON.parse('{{collection_id|js_string}}');
-      GLOBALS.collectionTitle = JSON.parse('{{collection_title|js_string}}');
-      GLOBALS.DEFAULT_TWITTER_SHARE_MESSAGE_PLAYER = JSON.parse(
-        '{{DEFAULT_TWITTER_SHARE_MESSAGE_PLAYER|js_string}}');
-      GLOBALS.explorationVersion = JSON.parse(
-        '{{exploration_version|js_string}}');
-      GLOBALS.INTERACTION_SPECS = JSON.parse(
-        '{{INTERACTION_SPECS|js_string}}');
-      GLOBALS.canEdit = JSON.parse('{{can_edit|js_string}}');
-      GLOBALS.username = JSON.parse('{{username|js_string}}');
-    </script>
-
-  {% endif %}
+  <script ng-if="version" type="text/javascript">
```

Comment:
Actually, just to check: is there any case where there wouldn't be a version here? I know the ng-if used to be there but I'm not sure I understand why we need it.

Reply:
I really can't comment on this. I myself am confused

Reply:
In terms of investigating, what have you tried, and what conclusions have you come to (or roadblocks have you encountered)? An important part of contributing to a codebase is reading code, not just writing it -- so, analysis of existing code is a very important skill. Otherwise you have no idea whether you're writing the correct code or not. It would be helpful if you could explain your reasoning in some detail so that we can help you get over the hump if you're stuck at a specific question.

Also, just FYI, if this issue/analysis is too hard, it's fine to close it and pick up a different issue instead.